### PR TITLE
Adding Postgres dashboards in metricbeat

### PR DESCRIPTION
**Pull request by dov0211**
Postgres dashboard added in metricbeat

### CODE REVIEW

**Comment by dov0211** [comment]
image

**Comment by ruflin** [comment]
Nice, thanks for the contribution. Any chance you could add this screenshot to the docs? See apache here for an example: https://github.com/elastic/beats/blob/master/metricbeat/module/apache/_meta/docs.asciidoc

The image file will belong here: https://github.com/elastic/beats/tree/master/metricbeat/docs/images

To check locally how the docs will look, you can run `make docs-preview`. It will take quite some time to build all the docs.

You also need to run `make update` in the metricbeat directory. The reason is that it will change the module overview pages to show that Postgres also has a dashboard 🎉

**Comment by ruflin** [comment]
@dov0211 Any updates on this one?

**Comment by dov0211** [comment]
@ruflin will look at this one this week

**Comment by dov0211** [comment]
image
@ruflin @jsoriano 
Let me know if this one is better

**Comment by dov0211** [comment]
image

**Comment by ruflin** [comment]
@jsoriano Could you check the above to see if this can be merged. Would be great to have a postgres dashboard.

**Comment by tbragin** [comment]
@dov0211 Thanks for working on this! I wonder if we can make titles on the dashboards more consistent from the spacing and capitalization perspective. Right now I see mix of first letter capitalized vs not capitalized, spaces around "-" and no spaces around "-"...

**Comment by ruflin** [comment]
@dov0211 Can we reactive this PR? Would be still great to get this dashboard in.

**Comment by dov0211** [comment]
yes, I should have the bandwidth working on that next week

**Comment by dov0211** [comment]
@ruflin - can you help me resolve the conflict?

**Comment by ruflin** [comment]
@dov0211 Running `make update` should do the trick.

**Comment by dov0211** [comment]
@ruflin still got some errors

**Comment by ruflin** [comment]
@dov0211 You got some errors locally? Because on Github no new version was pushed.

**Comment by ruflin** [comment]
@jsoriano Could you take a look at this again. @dov0211 I assume it's ready for review now.

File: metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json [hunk 1]
```diff
@@ -0,0 +1,737 @@
...
+  "objects": [
+    {
+      "attributes": {
+        "description": "",
+        "kibanaSavedObjectMeta": {
+          "searchSourceJSON": {
+            "filter": [],
+            "index": "metricbeat-*",
+            "query": {
+              "language": "lucene",
```
Comment (jsoriano) [inline metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json]: Query language should be `kuery` now, see #11268

File: metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json [hunk 2]
```diff
@@ -0,0 +1,737 @@
...
+                "extended_bounds": {},
+                "field": "@timestamp",
+                "interval": "auto",
+                "min_doc_count": 1,
+                "timeRange": {
+                  "from": "now-1h",
+                  "mode": "quick",
+                  "to": "now"
+                },
+                "time_zone": "Asia/Jerusalem",
```
Comment (jsoriano) [inline metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json]: Not sure why this is here, timezones shouldn't be stored in the dashboard.

File: metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json [hunk 3]
```diff
@@ -0,0 +1,737 @@
...
+                  "index": "metricbeat-*",
+                  "key": "postgresql.activity.query",
+                  "negate": true,
+                  "params": {
+                    "query": "",
+                    "type": "phrase"
+                  },
+                  "type": "phrase",
+                  "value": ""
+                },
```
Comment (jsoriano) [inline metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json]: I see in the screenshot that there is a filter for empty queries, when does it happen? Should we use this filter only in the visualization of recent queries?

File: metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json [hunk 4]
```diff
@@ -0,0 +1,737 @@
...
+              "y": 0
+            },
+            "id": "508fb7b0-f3bb-11e8-a48e-69084fbf7c82",
+            "panelIndex": "6",
+            "type": "visualization",
+            "version": "6.5.1"
+          }
+        ],
+        "timeRestore": false,
+        "title": "[Metricbeat Postgresql] overview",
```
Comment (jsoriano) [inline metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json]:
```suggestion
        "title": "[Metricbeat Postgresql] Overview",
```

File: metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json [hunk 5]
```diff
@@ -0,0 +1,737 @@
...
+          "searchSourceJSON": {
+            "filter": [],
+            "index": "metricbeat-*",
+            "query": {
+              "language": "lucene",
+              "query": ""
+            }
+          }
+        },
+        "title": "[Postgresql] Number of deadlock over time",
```
Comment (jsoriano) [inline metricbeat/module/postgresql/_meta/kibana/7/dashboard/Metricbeat-Postgresql-overview.json]: Visualization titles of this dashboard should have `[Metricbeat Postgresql]` as suffix
```suggestion
        "title": "Number of deadlock over time [Metricbeat Postgresql]",
```

**Comment by dov0211** [comment]
image

**Comment by jsoriano** [comment]
It LGTM, but tests in failures are related.

We test dashboards setup with the versions of the stack in metricbeat dockerfiles, and there we are still using 6.6.0. Dashboard in this PR were generated with 7.0 and they fail to be imported in 6.6 with:
```
2019-04-18T13:55:33.549Z	ERROR	instance/beat.go:856	Exiting: Failed to import dashboard: Failed to load directory /go/src/github.com/elastic/beats/metricbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/7/dashboard:
  error loading /go/src/github.com/elastic/beats/metricbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/7/dashboard/Metricbeat-postgresql-overview.json: returned 400 to import file: <nil>. Response: {"statusCode":400,"error":"Bad Request","message":"Document \"63b041e4-61ce-11e9-91d0-efd192a57cbc\" has property \"dashboard\" which belongs to a more recent version of Kibana (7.0.0)."}
Exiting: Failed to import dashboard: Failed to load directory /go/src/github.com/elastic/beats/metricbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/7/dashboard:
  error loading /go/src/github.com/elastic/beats/metricbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/7/dashboard/Metricbeat-postgresql-overview.json: returned 400 to import file: <nil>. Response: {"statusCode":400,"error":"Bad Request","message":"Document \"63b041e4-61ce-11e9-91d0-efd192a57cbc\" has property \"dashboard\" which belongs to a more recent version of Kibana (7.0.0)."}
```
I have opened https://github.com/elastic/beats/pull/11883 to try to update the dockerfiles to 7.0.

Not sure if there would be a way to convert it to the 6.X format in the meantime.

**Comment by jsoriano** [comment]
@dov0211 issue with 7.0 should be fixed now, could you try to update this branch with master?

**Comment by jsoriano** [comment]
Dashboard for Postgresql added in #13187, closing this one.

**Pull request closed without merging**